### ISSUE TITLE
Bake BlendShapes from current BlandShape Values

### ISSUE THREAD

**Issue by kazu0617** [Neos-Metaverse/NeosPublic#3314]
### Is your feature request related to a problem? Please describe.

現在、ブレンドシェイプを複数個細やかに動作させる場合、その値をそれぞれ保持しておく必要がありますが、これはパフォーマンス的に負荷を与え、またツール化する際の大きな妨げになっています。
これを解決するにはBlenderを使用してそれぞれのシェイプキーを指定の値に設定後、「Add Shape from Mix」をクリックして作成後、データをFBX等で出力し、Neosにインポートしなおすことで反映されますが、Neos内で直接操作することに比べ外部ツールを使用せねばならず、またBlenderの知識も必要です。また、AnimJを使用することもできますがさらに困難です。

### Describe the solution you'd like

> 「Skinned Mesh Renderer」のBlendShapeの項目に、[Add Shape from Mix]を追加し、クリックした際に（レガシーダイアログで構わないとは思います）以下のようなダイアログを出してください、その後[OK]をクリックすることで追加されてほしいです。

I would like to add [Add Shape from Mix] to the BlendShape section of the Skinned Mesh Renderer, and when I click on it (I think the legacy dialog is fine), I would like the following dialog to appear. I would like it to be added.
```
 -------------------- [ 📍 ] [ × ]
|        Blendshape Name:        |
| [                            ] |
| [      OK     ] [   Cancel   ] | 
 --------------------------------


 -------------------- [ 📍 ] [ × ]
|      Blendshape is Blank:      |
| [                            ] |
| [      OK     ] [   Cancel   ] | 
 --------------------------------
Blendshape Name: Inspector.Mesh.BlendshapeNameDialog
Blendshape is Blank: Inspector.Mesh.BlendshapeNameDialogWarning
OK: General.OK
Cancel: General.Cancel
```

### Describe alternatives you've considered

> - Blenderの機構を確認し、同様の動作をするようにライブラリを作成し、それらをプラグインかModでロードする。確認はしてみますが膨大な努力が必要かもしれません（C#は詳しくないです）。
> - Blenderを使用する。自分自身のデータでない場合で、かつ所持していないデータの場合は受け渡しを行うことは難しいため、Blenderの画面を共有しつつやり方を教える必要があり、非常に大変です（とはいえ、現在はこれが唯一の方法なので、それらに関する動画を作成するのも有効かもしれませんね）。

- Check Blender's mechanism, create libraries to work similarly, and load them with plugins or mods. I'll check it out, but it may require a huge amount of effort (I'm not familiar with C#).
- Use Blender. If the data is not your own and you don't own it, it is difficult to pass it on, so you need to teach them how to do it while sharing the Blender screen, which is very hard (although, this is the only way to do it now, so making a video about it might help! ).

### Additional context

I made an ASCII art of the dialog. Surprisingly, the width doesn't match.

**Comment by Frooxius** [comment]
I'm sorry, but I do not understand what exactly is this supposed to do. Can you provide me some examples of what does this operation do please?

**Comment by kazu0617** [comment]
OK. I'll write later, temporary post for Blender: https://docs.blender.org/manual/en/latest/animation/shape_keys/introduction.html#adding-shape-keys

**Comment by kazu0617** [comment]
I created a video, but only talk in japanese, so add sub here.

This section briefly explains how to create a blendshape.
"Specifying the shape key:
 Eyes_Thinner is set to 0.5.
 Mouth_Open_Smile2 is set to 1.
 Mouth_Angry is set to 0.6.

If you want to make this in Neos, you need to control each value with LogiX and components.
If you are in Blender, you can use this "New Shape From Mix" to create a new key.
The one created here is [Key 76].

If you check this shape key, you will see that the three shapes you specified above have been properly merged.
We want to do this within Neos.

ブレンドシェイプの作成方法について簡単に説明します。
「シェイプキーの指定」:
 Eyes_Thinner を 0.5 にセットしました。
 Mouth_Open_Smile2を 1 にセットしました。
 Mouth_Angry を 0.6 にセットしました。

これをNeosで作ろうと思うと、それぞれの値をLogiXやコンポーネントで制御する必要があります。
Blenderの場合であれば、この「New Shape From Mix」を使用して新たにキーを作成することができます。
ここで作成されたものが [Key 76] です。

このシェイプキーを確認すると、上で指定した3つのシェイプが適切にマージされています。

この作業をNeos内でやりたい、という要望です。

https://user-images.githubusercontent.com/6259214/140496303-f8bbadf9-7920-4b16-a36f-91c3c1cc6d25.mp4

**Comment by Frooxius** [comment]
Thank you for the information!

If I understand this correctly, it essentially merges multiple blendshapes into a single blendshape?

We can add this functionality in Neos, but it will require some changes to the inspector UI to support it first.

**Comment by kazu0617** [comment]
> 
> 
> Thank you for the information!
> 
> If I understand this correctly, it essentially merges multiple blendshapes into a single blendshape?
> 
> We can add this functionality in Neos, but it will require some changes to the inspector UI to support it first.

Yes, actually need to check the current size **as a percentage** before merging, but it's basically what you wrote.
